mountpoint-s3/tests/fuse_tests/write_test.rs

```diff
@@ -494,3 +494,53 @@ fn write_file(path: impl AsRef<Path>) -> std::io::Result<()> {
 fn write_with_invalid_storage_class_test_s3(storage_class: &str) {
     write_with_invalid_storage_class_test(crate::fuse_tests::s3_session::new, storage_class);
 }
+
+fn flush_test<F>(creator_fn: F, prefix: &str, append: bool)
```

Comment:
Another useful test (that maybe we should have written the first time we dealt with `flush`-related weirdness) would be one that literally just shells out to `touch /tmp/mount/file-that-doesnt-exist` and makes sure it doesn't fail. Don't do it if it's too hard, but might be helpful.